File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.py
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        nums.sort()
        res = []
        for q in queries:
            cnt = 0
            curq = q
            for n in nums:
                if n <= curq:
                    curq -= n
                    cnt += 1
                else:
                    break
            res.append(cnt)
        return res
            

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @return {number[]}
 */
var answerQueries = function(nums, queries) {
    nums.sort((a, b) => a - b)
    const res = []
    for (let q of queries) {
        let cnt = 0
        for (const n of nums) {
            if (n <= q) {
                cnt++
                q -= n
            } else {
                break
            }
        }
        res.push(cnt)
    }
    return res
};